src/main/java/de/tum/p2p/api/netty/forwarding/OnionTunnelDataMessage.java

```diff
@@ -0,0 +1,39 @@
+package de.tum.p2p.api.netty.forwarding;
+
+import de.tum.p2p.onion.forwarding.TunnelId;
+import de.tum.p2p.proto.message.MessageType;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.nio.ByteBuffer;
+
+@ToString
+@EqualsAndHashCode(callSuper = true)
+public class OnionTunnelDataMessage extends OnionApiMessage {
+
+    private final TunnelId tunnelId;
+    private final ByteBuffer data;
+
+    public OnionTunnelDataMessage(TunnelId id, ByteBuffer data) {
+        super(MessageType.ONION_TUNNEL_DESTROY);
+        this.tunnelId = id;
+        this.data = data;
+    }
+
+    public TunnelId getTunnelId() {
+        return tunnelId;
+    }
+
+    public ByteBuffer getData() { return data; }
```

Comment:
btw you can use `@Getter` class (applies to all fields) or field annotation for getters